pcdsdevices/pseudopos.py

```diff
@@ -282,6 +309,314 @@ def inverse(self, real_pos):
+        e.g. sync_limits = (-100, 100) will bind the `SyncAxis` to move between
+        -100 and 100.
+    """
+    sync = Cpt(PseudoSingleInterface, kind='omitted')
+
+    tab_whitelist = ['fix_sync']
+
+    # Enum that defines how and when offsets are interpreted
+    offset_mode = SyncAxisOffsetMode.STATIC_FIXED
+    # dict {str: float} that defines what offset to apply to each motor
+    offsets = None
+    # float that defines what offset to use for motors missing from offsets
+    default_offset = 0
+    # dict {str: float} that defines what scale to apply to each motor
+    scales = None
+    # float that defines what scale to use for motors missing from scales
+    default_scale = 1
+    # bool that defines if we should warn about inconsistent axes
+    warn_inconsistent = True
+    # float that defines how much of a deviation to warn on
+    warn_deadband = 0.001
+    # bool that defines which motor to keep still when calling fix_sync
+    # if omitted, we'll keep the first real motor still
+    fix_sync_keep_still = None
+    # tuple of two floats that defines if we should add limits to the sync axis
+    sync_limits = None
+
+    def __init__(self, *args, **kwargs):
+        self._check_settings()
+        super().__init__(*args, **kwargs)
+        self._real_attrs = [attr for attr, _ in self._get_real_positioners()]
+        self.scales = self._fill_info_dict(
+            self.scales, self.default_scale, 'scales')
+        if self.sync_limits is not None:
+            self.sync._limits = tuple(self.sync_limits)
+        self._has_setup = False
+
+    def _check_settings(self):
+        """Mostly just a typo check."""
+        if self.__class__ is SyncAxis:
+            raise TypeError(
+                'SyncAxis must be subclassed with '
+                'the axes to synchronize included as '
+                'components'
+                )
+        try:
+            self.offset_mode = SyncAxisOffsetMode(self.offset_mode)
+        except ValueError:
+            try:
+                self.offset_mode = SyncAxisOffsetMode[self.offset_mode]
+            except KeyError:
+                raise ValueError(
+                    f'Invalid offset_mode: {self.offset_mode}'
+                    ) from None
+        self._check_info_dict(self.offsets, 'offsets')
+        self._check_info_dict(self.scales, 'scales')
+        if self.fix_sync_keep_still is not None:
+            try:
+                getattr(self.__class__, self.fix_sync_keep_still)
+            except AttributeError:
+                raise ValueError(
+                    'Invalid fix_sync_keep_still. Must match a motor.'
+                    ) from None
+
+    def _check_info_dict(self, setting, info_kind):
+        """Helper function to check that all keys in the dict are Cpts"""
+        if setting is not None:
+            try:
+                for attr, _ in setting.items():
+                    getattr(self.__class__, attr)
```

Comment:
Perhaps `if attr not in self.component_names:` instead?

Reply:
I'll also do this